Riot/Managers/Room/RoomService.swift

```diff
@@ -0,0 +1,191 @@
+// 
+// Copyright 2020 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+
+/// `RoomService` errors
+enum RoomServiceError: Error {
+    case roomNotFound
+    case roomAlreadyJoined
+    case cannotJoinRoom
+    case roomAlreadyLeft
+    case cannotLeaveRoom
+    case changeMembershipAlreadyInProgress
+}
+
+/// `RoomService` enables to manage room
+@objcMembers
+class RoomService: NSObject, RoomServiceProtocol {
```

Comment:
This module does not update `RoomChangeMembershipStateDataSource` if the membership is changed from another matrix client;